### PR TITLE
fix: add redhat registry logo

### LINKED ISSUE #38: Add logo for Red Hat Registry
I'd love to see the Red Hat logo in the Registries menu after a successful login.  As shown below, there is a visual difference between the preset registries and the one added by the extension.  I'd like to fill this gap.

I am also looking into figuring out an alternative name to replace the URL `registry.redhat.io` with a more human-friendly name.

Screenshot 2024-02-13 at 13 46 12

Ignore until https://github.com/redhat-developer/podman-desktop-redhat-account-ext/pull/35 has merged.

Cc: @dgolovin @benoitf @jeffmaury @slemeur

### PR DESCRIPTION
**Pull request by dgolovin**
Fix #38.

image

### CODE REVIEW

**Comment by dgolovin** [comment]
There seems to be an issue with spacing between icon and registry name for this kind of entries.

**Review by benoitf — COMMENTED** [review]
IMHO we should include the png image as a resource

like https://github.com/containers/podman-desktop/blob/main/extensions/registries/src/extension.ts#L22

and https://github.com/containers/podman-desktop/blob/main/extensions/registries/src/extension.ts#L70-72

so you don't need to read the file, it's included in the .js file
also if the file is not found, it raises compilation error while in your case it'll be deferred to the runtime

**Comment by dgolovin** [comment]
I was looking for that, but missed it. I'll update the PR.

**Comment by vrothberg** [comment]
> There seems to be an issue with spacing between icon and registry name for this kind of entries.

Is that something to fix in Podman Desktop or the extension?

Is there a way we can change the name to "Red Hat Container Registry"?

**Review by jeffmaury — COMMENTED** [review]
Shouldn't we have a more friendly name ?

File: src/extension.ts [hunk 1]
```diff
@@ -75,7 +83,8 @@ async function createRegistry(username: string, secret: string, serverUrl: strin
     serverUrl,
     username,
     secret,
-    source: ''
+    source: '',
```
Comment (benoitf) [inline src/extension.ts:86]:
```suggestion
    source: '',
    name: 'Red Hat Container Registry',
```

**Comment by benoitf** [comment]
I've added the name change in the GitHub UI suggested change

**Comment by vrothberg** [comment]
> I've added the name change in the GitHub UI suggested change

Looks great, thank you!
image

The only very minor thing that strikes my eyes is the user name.  It looks cryptic as it's machine-generated.  Not sure how we could beautify that.

**Comment by benoitf** [comment]
- do you have a way to provide metadata to make it shorter as well ? (like -pd vs -podman-desktop) (probably something we can tune on our side)
- can we have emails rather than digits on registry.redhat.io side ? like username "valentin@redhat.com"

- we could enrich Podman Desktop API to include a username alias. So it could display "valentin@redhat.com" but if you try to edit it'll display you real digit-podman-desktop value but at least it would render more nicely. One issue is that for now the data are stored in auth.json so there won't be issue in Podman if we add extra fields in auth.json ?

like
```json
{
        "auths": {
                "quay.io": {
                        "alias": "fbenoit@redhat.com",
                        "auth": "encoded-value"
                }
        }
}
```

**Comment by vrothberg** [comment]
I am unable to answer the questions but am sure that @dgolovin knows.  I like the idea of displaying the email address used in SSO.

**Comment by benoitf** [comment]
@vrothberg do you know if podman will ignore extra fields that we may store in `containers/auth.conf` file ?

**Comment by vrothberg** [comment]
> @vrothberg do you know if podman will ignore extra fields that we may store in `containers/auth.conf` file ?

It's JSON, so it very likely ignores unknown fields.  But it's dangerous territory as it may break at any time in the future if we exploit the auth.conf for another purpose.

**Review by benoitf — APPROVED** [review]
I'm also fine to have the png import in a follow up PR

**Comment by dgolovin** [comment]
@benoitf There is no way to control user name, I call API with request to create 'podman-desktop' service account and it returns json with username/password. It looks like user name is generated as ${organizationID}|${requested-service-account-name}.

**Comment by dgolovin** [comment]
BTW icon and name for the registry does not survive podman desktop restart https://github.com/containers/podman-desktop/issues/6218

**Comment by dgolovin** [comment]
Created follow up  issue https://github.com/redhat-developer/podman-desktop-redhat-account-ext/issues/68

**Pull request merged**